linode/manager
# test: [M3-7367] - Improve Cypress Rescue & Rebuild Test Flake

## Description 📝
This PR addresses some of the causes of flakiness and failures in our Linode rebuild and rescue Cypress tests.

## Changes  🔄
- Rescue test:
  - Fixed an issue on reattempts where the test would attempt to create a Linode with the same label as in the previous attempt, leading to a consistent Cypress failure stemming from a 400 response from the API on all reattempts
  - Added resource clean up
  - Refactored the _Cannot reboot a provisioning Linode into rescue mode_ test to use mock API data
- Rebuild test:
  - Addressed primary cause of flakiness by navigating directly to Linode details page and initiating rebuild from there
  - Improve resource clean up by deleting LKE clusters in addition to other resources
  - Refactor tests to use latest patterns, fewer markup-dependent selectors, etc.
  - Refactored the _cannot rebuild a provisioning linode_ to use mock API data
- Other supporting changes:
  - Mock user preference requests in Linode landing table _checks summary view for linode table_ test to decrease the risk of user preference changes interfering with subsequent tests
  - Add intercept and mock utils, UI helpers

## How to test 🧪

### Prerequisites
Build and serve Cloud Manager:
```bash
yarn && yarn build && yarn start:manager:ci
```

### Verification steps 
Make sure that the tests that were modified by this PR continue to pass. We can rely on CI for this, but to run the tests manually you can use this command:
```bash
yarn cy:run -s "cypress/e2e/core/linodes/linode-config.spec.ts,cypress/e2e/core/linodes/rebuild-linode.spec.ts,cypress/e2e/core/linodes/rescue-linode.spec.ts,cypress/e2e/core/linodes/smoke-linode-landing-table.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*
